pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -10,18 +10,17 @@
   libICE,
   libSM,
   libX11,
-  nexusmods-app,
   runCommand,
   pname ? "nexusmods-app",
 }:
-buildDotnetModule rec {
+buildDotnetModule (finalAttrs: {
   inherit pname;
   version = "0.4.1";
 
   src = fetchFromGitHub {
     owner = "Nexus-Mods";
     repo = "NexusMods.App";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
```

Comment:
This is interesting.  Is this a common pattern?

I think I might actually be surprised if overriding the version automatically changed `src`, but it does seem like it could be useful anywhere `rec` is currently used...

Reply:
> Is this a common pattern?

I'm pretty new to packaging stuff, so really not the best person to ask, but just [searching the repo](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20finalAttrs.version&type=code) does bring up quite a few hits.

Maybe someone like @eclairevoyant would have an opinion?

> I think I might actually be surprised if overriding the version automatically changed `src`

I'm not sure. I can see where you're coming from, but is it common to override the `version` without actually downloading a different `src`? What would be the use-case?

Regardless, I'd like to take the time to get it right in this PR, so all feedback welcome!

Reply:
> There are many code adopting `finalAttrs` to reuse variables, rather than provide an overridable option. Specifically, using `finalAttrs.version` and `finalAttrs.pname` in `src = ` is incorrect since overriding them will NOT cause it to automatically use the source from new name/version, because the hash literal is still unchanged.

#310373 seems to cover this issue, I guess we should continue using `rec` or `let in` for the version.

Reply:
Reverted the `version` to use `rec` again

Reply:
Hm, reading a bit more it seems the best practice is still up in the air?

e.g. https://github.com/NixOS/nixpkgs/issues/293452#issuecomment-1979152319 from @eclairevoyant:
> This isn't caused nor exacerbated by `finalAttrs` at all. If you override `version` and not `src`, even without `finalAttrs`, you'll also silently use the existing sources. That's just how FODs work.
> 
> The `finalAttrs` pattern in fact makes it easier to override `src`, as you _may_ only need to set `outputHash`

I'll un-revert back to using `finalAttrs` for now, pending further feedback and a consensus.

Reply:
`finalAttrs.pname` in a URL (and by extension, a fetcher) is objectively wrong. `pname` is our internal packaging attr, changing it should not break downloading sources.

However, `finalAttrs.version` is contentious, as it really should correspond to the upstream version (unless we're using an arbitrary commit, in which case you'd have to set `src.rev` manually anyway). No matter what is done (`let`, `rec`, or `finalAttrs`), you'll have to override `src` to get the right version; if you don't, you'll experience the well-known "FOD moment" in all 3 cases.